Zend/zend_object_handlers.c

```diff
@@ -1941,6 +1941,10 @@ ZEND_API HashTable *zend_get_properties_for(zval *obj, zend_prop_purpose purpose
 	return zend_std_get_properties_for(zobj, purpose);
 }
 
+ZEND_API zend_result zend_std_equals_objects(zend_uchar opcode, zval *result, zval *op1, zval *op2) {
+	return FAILURE;
```

Comment:
Can the fallback to the compare handler be in here?

Do we really need to pass the opcode here? I think it would be better for the return value to indicate whether it equals or not (possibly with an equals/not_equals/uncomparable enum). This would prevent inconsistent `==` and `!=` implementations by design.

Reply:
The reason I didn't put the fallback here is that extensions that used this handler would have to manually implement the fallback again (assuming they wanted to use the compare handler). But yeah, if this function returned a result enum instead of `zend_result` I think the fallback would *have* to go in here.

That's just very different from how the rest of the handlers are done (from what I've seen), and I wanted this code to be similar to other code in the engine.

Reply:
> The reason I didn't put the fallback here is that extensions that used this handler would have to manually implement the fallback again (assuming they wanted to use the compare handler).

Not sure I get what you mean here -- if they want to use some custom logic and then fall back to the default, they can call zend_std_equals_objects to do so.

Reply:
> Not sure I get what you mean here -- if they want to use some custom logic and then fall back to the default, they can call zend_std_equals_objects to do so.

Or yeah... they could do the obvious thing you just suggested...

Reply:
Actually though, what about the case where op1 falls back but op2 has custom logic? How would the default handler for op1 handle that?